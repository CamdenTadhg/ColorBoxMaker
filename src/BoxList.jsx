import react, {useState} from "react";
import './BoxList.css';
import NewBoxForm from './NewBoxForm.jsx';
import Box from './Box.jsx';
import {v4 as uuid} from 'uuid';

const BoxList = () => {
    const INITIAL_STATE = [
        {id: uuid(), color: 'blueviolet' , width: 18, height: 21},
        {id: uuid(), color: 'darkmagenta', width: 17, height: 17},
        {id: uuid(), color: 'darkorchid', width: 28, height: 23}
    ]
    const [boxList, setBoxList] = useState(INITIAL_STATE);

    const addBox = (color, width, height) =>{
        const boxListCopy = boxList.map(box => ({...box}))
        boxListCopy.push({id:uuid(), color, width, height});
        setBoxList(boxListCopy);
    };

    const deleteBox = (id) => {
        const newBoxList = boxList.filter((box) => box.id !== id);
        setBoxList(newBoxList);
    }

    return (
        <div className="BoxList">
            <NewBoxForm addBox={addBox}/>
            {boxList.map(b => (
                <Box key={b.id} color={b.color} width={b.width} height={b.height} deleteBox={() => deleteBox(b.id)}/>
            ))}
        </div>
    )
}

export default BoxList;