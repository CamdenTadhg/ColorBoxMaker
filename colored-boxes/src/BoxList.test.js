import react from 'react';
import {render} from '@testing-library/react';
import BoxList from './BoxList.js'

it('should render correctly', function() {
    render(<BoxList/>);
});

it('should match snapshot', function(){
    const {asFragment} = render(<BoxList/>);
    expect(asFragment()).toMatchSnapshot();
});