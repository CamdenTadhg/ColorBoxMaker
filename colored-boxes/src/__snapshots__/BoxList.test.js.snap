// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form
      class="NewBoxForm"
    >
      <label
        class="NewBoxForm-label"
        for="color"
      >
        Color:
        <input
          id="color"
          name="color"
          placeholder="color"
          type="text"
          value=""
        />
      </label>
      <label
        class="NewBoxForm-label"
        for="width"
      >
        Width:
        <input
          id="width"
          name="width"
          placeholder="width"
          type="number"
          value="0"
        />
      </label>
      <label
        class="NewBoxForm-label"
        for="height"
      >
        Height:
        <input
          id="height"
          name="height"
          placeholder="height"
          type="number"
          value="0"
        />
      </label>
      <button>
        Add Box
      </button>
    </form>
    <div
      class="Box"
      style="background-color: blueviolet; width: 18vw; height: 21vh;"
    />
    <button
      class="Box-button"
    >
      X
    </button>
    <div
      class="Box"
      style="background-color: darkmagenta; width: 17vw; height: 17vh;"
    />
    <button
      class="Box-button"
    >
      X
    </button>
    <div
      class="Box"
      style="background-color: darkorchid; width: 28vw; height: 23vh;"
    />
    <button
      class="Box-button"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
